cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/FormInputComponents/OsImageSelect/OsImageSelect.tsx

```diff
@@ -3,27 +3,31 @@ import React, { ChangeEvent } from 'react';
 
 // Library imports
 import { CdsControlMessage } from '@cds/react/forms';
-import { FieldError } from 'react-hook-form';
+import { useFormContext } from 'react-hook-form';
 import { CdsSelect } from '@cds/react/select';
 import * as yup from 'yup';
+import { AWSVirtualMachine } from '../../../../swagger-api';
 
-interface ImageProps<T> {
+interface ImageProps {
     osImageTitle: string;
     field: string;
-    errors: { [key: string]: FieldError | undefined };
-    register: any;
     onOsImageSelected: (osImage: string, fieldName?: string) => void;
-    images: T[];
+    images: AWSVirtualMachine[];
```

Comment:
I'm confused why this shared component would go from taking a generic image type to an AWS-specific image type.

Reply:
@swalner-vmware my IDE complained when it was generic. And I'm not sure that this is a good place to use generic